docs/content/concepts/resources.mdx

```diff
@@ -5,357 +5,353 @@ description: Resources enable you to separate graph logic from environment, and
+If your resource requires other resources, then you can pass them as constructor arguments.
 
-```python file=/concepts/resources/resources.py startafter=start_resource_testing_with_context endbefore=end_resource_testing_with_context
-from dagster import build_init_resource_context, resource
+```python file=/concepts/resources/pythonic_resources.py startafter=start_new_resource_testing_with_context endbefore=end_new_resource_testing_with_context dedent=4
+from dagster import ConfigurableResource
 
+class StringHolderResource(ConfigurableResource):
+    value: str
 
-@resource(required_resource_keys={"foo"}, config_schema={"bar": str})
-def my_resource_requires_context(init_context):
-    return init_context.resources.foo, init_context.resource_config["bar"]
-
+class MyResourceRequiresAnother(ConfigurableResource):
+    foo: StringHolderResource
+    bar: str
 
 def test_my_resource_with_context():
-    init_context = build_init_resource_context(
-        resources={"foo": "foo_str"}, config={"bar": "bar_str"}
-    )
-    assert my_resource_requires_context(init_context) == ("foo_str", "bar_str")
-```
-
-If your resource is a context manager, then you can open it as one using python's `with` syntax.
-
-```python file=/concepts/resources/resources.py startafter=start_test_cm_resource endbefore=end_test_cm_resource
-from contextlib import contextmanager
-from dagster import resource
-
-
-@resource
-@contextmanager
-def my_cm_resource(_):
-    yield "foo"
-
-
-def test_cm_resource():
-    with my_cm_resource(None) as initialized_resource:
-        assert initialized_resource == "foo"
+    string_holder = StringHolderResource(value="foo")
+    resource = MyResourceRequiresAnother(foo=string_holder, bar="bar")
+    assert resource.foo.value == "foo"
+    assert resource.bar == "bar"
 ```
 
 ---
 
-## Initializing resources outside of execution
+## Defining Pythonic I/O managers
 
-There are scenarios where you might want to reuse the code written within your resources outside of the context of execution. Consider a case where you have a resource `db_connection`, and you want to use that resource outside of the context of an execution. You can use the <PyObject object="build_resources"/> API to initialize this resource outside of execution.
+Pythonic [I/O managers](/concepts/io-management/io-managers) are defined as subclasses of <PyObject object="ConfigurableIOManager"/>, and similarly to Pythonic resources specify any configuration fields as attributes. Each subclass must implement a `handle_output` and `load_input` method, which are called by Dagster at runtime to handle the storing and loading of data.
 
-```python file=/concepts/resources/resources.py startafter=start_build_resources_example endbefore=end_build_resources_example
-from dagster import resource, build_resources
+If you are interested in adapting a legacy, function-styled I/O manager, see the [adapting function-style I/O managers section](#adapting-function-style-io-managers).
```

Comment:
Is the `function-styled I/O manager` section supposed to be on this page? I checked out the legacy page and didn't see it, so perhaps this was an oversight?

Reply:
Good catch, will offload that to the migration guide.